import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ImportAdapter.java

```diff
@@ -190,16 +186,22 @@ private static IDestinationClientFactory configureDestinationClientFactory(
           flags.destinationConfigInline,
           flags.destinationConfigPath,
           DestinationsConfig.ENV_DESTINATION_CONFIG_JSON,
-          credentials);
+          credentials, flags.useStowOverwrite);
 
       destinationClientFactory = new MultipleDestinationClientFactory(
           multipleDestinations.getLeft(),
           multipleDestinations.getRight(),
           defaultCstoreDicomWebClient);
     } else { // with or without backup usage.
+      if ((flags.destinationConfigPath != null || flags.destinationConfigInline != null)
+          && flags.useStowOverwrite) {
+        throw new IllegalArgumentException(
+            "Must use '--send_to_all_matching_destinations' when using '--stow_overwrite' and"
+                + " providing a destination config.");
+      }
```

Comment:
I looked in the cstore service to refresh my memory. So it looks like the condition for this to work is multipleDestinationSendService != null && useStowOverwrite = true, that way we take the path where we use the backup service, and can use the delete and re-store logic. [1] Just wondering why it matters if --send_to_all_matching_destinations is true vs. just using one destination here? 

Can we simplify this check to:
if multipleDestinationSendService == null && useStowOverWrite, then throw error.

[1] https://github.com/GoogleCloudPlatform/healthcare-dicom-dicomweb-adapter/blob/c893c51a4cce514e10511fb9d53f9205e60c2f17/import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java#L119

Reply:
Two reasons: 
1) The import adapter always [initializes](https://github.com/GoogleCloudPlatform/healthcare-dicom-dicomweb-adapter/blob/master/import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java#L129) the multipleDestinationSendService, so it won't be `null`.
2) This one took me a bit of time to understand, but this [code](https://github.com/GoogleCloudPlatform/healthcare-dicom-dicomweb-adapter/blob/master/import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ImportAdapter.java#L181) initializes the `destinationClientFactory` based on send_to_all_matching_destinations, and then the the `Single/MultipleDestinationClientFactory` will assign the destination to the `DestinationHolder` based on the C-store requests. The problem is the single/multiple implementations actually assign the values based on different properties and the values have to be retrieved via different properties (namely `getSingleDestination()` vs `getHealthcareDestinations()`). And then this [code](https://github.com/GoogleCloudPlatform/healthcare-dicom-dicomweb-adapter/blob/master/import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java#L129) relies on the `DestinationHolder` (though `multipleSendService` will never be null in import adapter mode).

Reply:
For 1. It looks like we do sometimes not initialize multipleDestinationSendService, if backupUploadService == null: 
https://github.com/GoogleCloudPlatform/healthcare-dicom-dicomweb-adapter/blob/195a0af70b1fed12df24575fad7b1375faff63d5/import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ImportAdapter.java#L220

For 2. hmm this is interesting. So it looks like there's a bug in the code? When:
- the single destination client factory is specified
- backup service is initialized
- but sendToAllMatchingDatasets is not specified

then we will use multipleDestinationSendService and call destinationHolder.getHealthcareDestinations() in CStoreService.Java.
However destinationHolder.getHealthcareDestinations() will not return what we expect, since single destination client factory only returns anything on getSingleDestination().

Reply:
The bug in 2. being that we would just be sending to all destinations rather than the single destination that we intended to send based on the filter. And the filtering in the single destination would not work. 

Am I correct? The code is a bit complex here so I could have got this wrong. Anyway I think that issue is outside of the scope of this PR, but I wanted to get on the same page with this before deciding what to do about it.

Reply:
As discussed, going to leave the code as-is for now. Created issue #144 for 2.